OWML.ModHelper.Interaction/InterfaceProxyBuilder.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text;
+
+namespace OWML.ModHelper.Interaction
+{
+    internal class InterfaceProxyBuilder
+    {
+
+        private readonly Type TargetType;
+
+        private readonly Type ProxyType;
+
+        /// <summary>Construct an instance.</summary>
+        /// <param name="name">The type name to generate.</param>
+        /// <param name="moduleBuilder">The CLR module in which to create proxy classes.</param>
+        /// <param name="interfaceType">The interface type to implement.</param>
+        /// <param name="targetType">The target type.</param>
+        public InterfaceProxyBuilder(string name, ModuleBuilder moduleBuilder, Type interfaceType, Type targetType)
+        {
+            // validate
+            if (name == null)
+                throw new ArgumentNullException(nameof(name));
+            if (targetType == null)
+                throw new ArgumentNullException(nameof(targetType));
+
+            // define proxy type
+            TypeBuilder proxyBuilder = moduleBuilder.DefineType(name, TypeAttributes.Public | TypeAttributes.Class);
+            proxyBuilder.AddInterfaceImplementation(interfaceType);
+
+            // create field to store target instance
+            FieldBuilder targetField = proxyBuilder.DefineField("__Target", targetType, FieldAttributes.Private);
+
+            // create constructor which accepts target instance and sets field
+            {
```

Comment:
This is more about my lack of C# knowledge, but any clue why this block starts here? I guess to separate the scope, but seems weird to me. Maybe it should be a separate method. Also, separating some of this stuff into individual methods would make for easier debugging and less comments needed.